cloud/amazon/ec2_ami.py

```diff
@@ -381,8 +467,13 @@ def create_image(module, ec2):
                 bd = BlockDeviceType(**device)
                 bdm[device_name] = bd
             params['block_device_mapping'] = bdm
+            register_params['block_device_mapping'] = bdm
+
+        if instance_id:
```

Comment:
you are right. This check should really be checking to see what the action is (create or register) I will update it tonight.